### ISSUE TITLE
Query with 'top' statement is not SQLSever compatible

### ISSUE THREAD

**Issue by StefanBollmannCAS** [SAP/olingo-jpa-processor-v4#416]
As [described on SO](https://stackoverflow.com/questions/79214476/spring-boot-3-with-olingo-jpa-processor-v4-sqlserver-error-incorrect-syntax-n) an upgrade from olingo 1.1.1 to 2.1.3 while upgrading to Spring Boot 3 gives now in a SQL Server application SQL statements like this:
```
SELECT E0.ContactId S0 FROM T_Case E0 ORDER BY E0.InternalCaseId ASC LIMIT 10
```
In Spring Boot 2.7.3 with olingo 1.1.1 the SQL statement was
```
SELECT E0.ContactId S0 FROM T_Case E0 ORDER BY E0.InternalCaseId DESC
```

SQL Server does not understand 'LIMIT'. So we get 

```
{
	"error": {
		"code": null,
		"message": "Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\nInternal Exception: com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near 'LIMIT'.\nError Code: 102\nCall: SELECT E0.CaseNumber S0 FROM T_Case E0 ORDER BY E0.InternalCaseId ASC LIMIT 5\")"
	}
}
```

**Comment by wog48** [comment]
This seems to be a similar issue as [307](https://github.com/SAP/olingo-jpa-processor-v4/issues/307) solved with 2.2.0.

SQLServer does not support LIMIT and OFFSET in a way e.g. H2 or PostgeSQL. Instead `OFFSET n ROWS` and `FETCH NEXT n ROWS ONLY` have to be used.
Beside differences with the pagination clause, there are differences with String function. Known are: 

- [LEN](https://learn.microsoft.com/en-us/sql/t-sql/functions/len-transact-sql?view=sql-server-ver16) - JPA Processor uses `LENGTH`
- [CHARINDEX](https://learn.microsoft.com/en-us/sql/t-sql/functions/charindex-transact-sql?view=sql-server-ver16) - JPA Processor uses `LOCATE`

As other databases differ in other functions, it is necessary to have a database specific mapping, which was implemented in 2.2.0.

Quick solution is to remove the dependency to module [odata-jpa-processor-cb](https://github.com/SAP/olingo-jpa-processor-v4/tree/main/jpa/odata-jpa-processor-cb), which may come with some performance drawbacks. It is also an option to upgrade to Spring Boot 3.x.x, as 2.x.x is out of maintenance, together with an update to JPA Processor 2.3.0

**Comment by StefanBollmannCAS** [comment]
> As other databases differ in other functions, it is necessary to have a database specific mapping, which was implemented in 2.2.0.
Thank you for this hint @wog48. Yes, we tried with 2.3.0. When we send a request with "top" query this issue described here does not happen anymore.
> It is also an option to upgrade to Spring Boot 3.x.x, as 2.x.x is out of maintenance, together with an update to JPA Processor 2.3.0
We are in the process to upgrade. So, Spring Boot 3.2.5 is already used.
